### DISCUSSION TITLE
Angular support?

### DISCUSSION THREAD

**Discussion by iCodr8** [tailwindlabs/heroicons#514] [Ideas]
Why is there no angular support? Or will it be implemented soon?

**Comment by reinink** [comment]
Hey there! Moving this over to discussions, as it's more of a feature request than an issue.

That said, I sort of doubt we'll ever add direct Angular support to Hericons, as we don't currently offer direct Angular support with any of our other projects (Headless UI, Tailwind UI).

That said, while I've never worked with Angular personally, I have to believe that Angular has some way to render a regular `<svg>` element, no? Meaning you can just copy the SVG code from the Hericons website and use that in your Angular project 👍

**Reply by jderboven** [reply]
That would have been fun to be able to use it like any other fonts (font-awesome or google-material) within angular.

Actually yes you can inject directly the svg into HTML but when you have to do it dynamically, it gets messy as of Angular has security concerns about tags such as svg. Having a lib such as @heroicons/angular might be a good idea then

**Reply by brandonpittman** [reply]
I created a private library to handle this at my old job. It’s doable, just not pleasant.

**Comment by ryaa** [comment]
To whom it may concern...

If you need to consume a heroicon icon in your ionic/angular project you can follow the below steps:
Pre-requisites:
- change **angular.json** file and add the below lines under **projects --> app --> architect --> build --> options --> assets** (where **src/assets/icon/heroicons** is the path where you will be saving svg icon files - see the step 1 below)
```
    {
      "glob": "**/*.svg",
      "input": "src/assets/icon/heroicons",
      "output": "./svg"
    }
```

To add an icon please:
1) create an svg file in **<Git>/src/assets/icon/heroicons** with the heroicon name (for example, **building-library**) and past this icon svg in this file
2) add the below to the global (if the icon used in more then one component) or to an individual component scss file (please use the svg file name for the `name` and `url` as shown below)
```
ion-icon {
  &[name*="building-library"] {
    mask-image: url(/svg/building-library.svg);
  }
}
```
3) add the **ion-icon** with **name** attribute having the icon name, for example:
```
<ion-icon name="building-library"></ion-icon>
```

**Comment by quedicesebas** [comment]
You can use https://www.npmjs.com/package/@ng-icons/ionicons
